### ISSUE TITLE
XSS vulnerability

### ISSUE THREAD

**Issue by JinYiTong** [gongfuxiang/shopxo#57]
After the template file of ‘index.php?s=order/index' is parsed, there is a controllable variable here.
![image](https://user-images.githubusercontent.com/71314272/128523145-a037569c-53b9-40b1-b22d-5cbcfb558513.png)
![image](https://user-images.githubusercontent.com/71314272/128522956-ccac2f1f-2621-4f60-b229-b195b2714e1e.png)
By using payload `index.php?s=order&ids="><script>alert(1);</script>` ,we can get an xss alert.But it need anyone is logining,so hacker can get user's cookies when they visit the evil link.
![image](https://user-images.githubusercontent.com/71314272/128523561-051ce376-a044-48fa-bd3c-b6e4dd0db81d.png)

**Comment by gongfuxiang** [comment]
> 'index.php?s=order/index'的模板文件解析后，这里有一个可控变量。 通过使用payload ，我们可以获得一个xss警报。但是它需要任何人登录，所以黑客可以在访问恶意链接时获取用户的cookie。
> ![图像](https://user-images.githubusercontent.com/71314272/128523145-a037569c-53b9-40b1-b22d-5cbcfb558513.png)
> ![图像](https://user-images.githubusercontent.com/71314272/128522956-ccac2f1f-2621-4f60-b229-b195b2714e1e.png)
> `index.php?s=order&ids="><script>alert(1);</script>`
> ![图像](https://user-images.githubusercontent.com/71314272/128523561-051ce376-a044-48fa-bd3c-b6e4dd0db81d.png)

您好、非常感谢您的反馈，这个是新版本框架升级忘记开启参数验证了，前几天已在dev分支修复。

**Comment by JinYiTong** [comment]
Yeah,you hava fixed it.
![image](https://user-images.githubusercontent.com/71314272/128585667-c17ef5b6-ba19-4e21-8c7e-6a7aa228d4dd.png)

**Comment by JinYiTong** [comment]
No,No,No.This vulnerability still exists. You only use htmlspecialchars for `data_request`, and then url-decode `$params['ids']`, so we can bypass it using the second url encoding.
`?s=order&ids=%2522%253E%253Cscript%253Ealert(1)%3B%253C%2Fscript%253E`
![image](https://user-images.githubusercontent.com/71314272/128588680-ce728682-dedf-4066-8eb1-5f10768ff7b3.png)

**Comment by gongfuxiang** [comment]
> No,No,No.This vulnerability still exists. You only use htmlspecialchars for `data_request`, and then url-decode `$params['ids']`, so we can bypass it using the second url encoding.
> `?s=order&ids=%2522%253E%253Cscript%253Ealert(1)%3B%253C%2Fscript%253E`
> ![image](https://user-images.githubusercontent.com/71314272/128588680-ce728682-dedf-4066-8eb1-5f10768ff7b3.png)

您好，非常感谢您的深度测试反馈，刚才已在 dev 分支提交了新的修复方案，如有问题可以再次提交记录。
